sdk/core/azure-core/test/ut/azure_libcurl_core_main.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file This test assumes the application is already using libcurl.
+ *
+ */
+
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+
+#include <gtest/gtest.h>
+
+#include <curl/curl.h>
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/curl_transport.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/io/body_stream.hpp>
+#include <azure/core/response.hpp>
+
+#include <http/curl/curl_connection_pool_private.hpp>
+#include <http/curl/curl_connection_private.hpp>
+#include <http/curl/curl_session_private.hpp>
+
+#include <cstdlib>
+
+namespace Azure { namespace Core { namespace Test {
+  TEST(SdkWithLibcurl, globalCleanUp)
+  {
+    Azure::Core::Http::Request req(
+        Azure::Core::Http::HttpMethod::Get, Azure::Core::Url("https://httpbin.org/get"));
+
+    {
+      // Creating a new connection with default options
+      Azure::Core::Http::CurlTransportOptions options;
+      auto connection
+          = Azure::Core::Http::_detail::CurlConnectionPool::ExtractOrCreateCurlConnection(
+              req, options);
+
+      auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+          req, std::move(connection), options.HttpKeepAlive);
+      auto result = session->Perform(Azure::Core::Context::GetApplicationContext());
+      (void)result;
+      // Reading all the response
+
+      session->ReadToEnd(Azure::Core::Context::GetApplicationContext());
+    }
+    // Check that after the connection is gone, it is moved back to the pool
+    EXPECT_EQ(Azure::Core::Http::_detail::CurlConnectionPool::ConnectionPoolIndex.size(), 1);
+    auto connectionFromPool
+        = Azure::Core::Http::_detail::CurlConnectionPool::ConnectionPoolIndex.begin()
+              ->second.begin()
+              ->get();
+    (void)connectionFromPool;
+  }
+}}} // namespace Azure::Core::Test
+
+int main(int argc, char** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
```

Comment:
Did you intend to call curl_global_init() and curl_global_cleanup() in this main function or in the test case itself?

Reply:
No, I explicitly removed all the calls we used to have to curl_global_init and curl_global_cleanup from the test applications.
The SDK should now call it during the static init and during app exit.